crates/astria-sequencer/src/slinky/oracle/component.rs

```diff
@@ -0,0 +1,71 @@
+use std::sync::Arc;
+
+use anyhow::{
+    Context,
+    Result,
+};
+use astria_core::{
+    protocol::genesis::v1alpha1::GenesisAppState,
+    slinky::oracle::v1::CurrencyPairState,
+};
+use tendermint::abci::request::{
+    BeginBlock,
+    EndBlock,
+};
+use tracing::instrument;
+
+use super::state_ext::StateWriteExt;
+use crate::component::Component;
+
+#[derive(Default)]
+pub(crate) struct OracleComponent;
+
+#[async_trait::async_trait]
+impl Component for OracleComponent {
+    type AppState = GenesisAppState;
+
+    #[instrument(name = "OracleComponent::init_chain", skip(state))]
+    async fn init_chain<S: StateWriteExt>(mut state: S, app_state: &Self::AppState) -> Result<()> {
+        for currency_pair in &app_state.slinky().oracle().currency_pair_genesis {
+            let currency_pair_state = CurrencyPairState {
+                id: currency_pair.id(),
+                nonce: currency_pair.nonce(),
+                price: currency_pair.currency_pair_price().clone(),
+            };
+            state
+                .put_currency_pair_state(currency_pair.currency_pair(), currency_pair_state)
+                .context("failed to put currency pair")?;
+            tracing::info!(
```

Comment:
Remove this here. This information will be available form the span created by `put_currency_pair_state`